### PR TITLE
HDDS-6018 Closing state to quasiclosed

### PR DESCRIPTION
**Pull request by kerneltime**
## What changes were proposed in this pull request?

If a container is in closing state, the next state for it should be
quasi closed for containers without a pipeline.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-6018

## How was this patch tested?

- [x] Integration tests

### CODE REVIEW

**Comment by guihecheng** [comment]
Oh this should a good fix, we've encountered a lot of suspicious UNHEALTHY containers in our clusters. @ChenSammi

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java [hunk 1]
```diff
@@ -78,6 +77,7 @@ public void handle(SCMCommand command, OzoneContainer ozoneContainer,
         ((CloseContainerCommand)command).getProto();
     final ContainerController controller = ozoneContainer.getController();
     final long containerId = closeCommand.getContainerID();
+    LOG.debug("Processing Close Container command containerID={}", containerId);
```
Comment (swagle) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java:80]: Minor nit: wrap in an if check, debug/trace logging params are evaluated even if the log level is disabled.

Comment (kerneltime) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java:80]: Will do

**Comment by kerneltime** [comment]
Question: We should also be changing assumption here: https://github.com/apache/ozone/blob/master/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/AbstractContainerReportHandler.java#L195
It is possible that the container has changed state while SCM missed a few heart beats?

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java [hunk 2]
```diff
@@ -78,6 +77,10 @@ public void handle(SCMCommand command, OzoneContainer ozoneContainer,
         ((CloseContainerCommand)command).getProto();
     final ContainerController controller = ozoneContainer.getController();
     final long containerId = closeCommand.getContainerID();
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Processing Close Container command containerID={}",
```
Comment (bharatviswa504) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java:81]: Minor: space between = or we can use ":" like other places

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java [hunk 3]
```diff
@@ -127,7 +128,9 @@ public void handle(SCMCommand command, OzoneContainer ozoneContainer,
         break;
       }
     } catch (NotLeaderException e) {
-      LOG.debug("Follower cannot close container #{}.", containerId);
+      if (LOG.isDebugEnabled()) {
```
Comment (bharatviswa504) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java:131]: I don't think we need isDebugEnabled

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java [hunk 4]
```diff
@@ -78,6 +77,10 @@ public void handle(SCMCommand command, OzoneContainer ozoneContainer,
         ((CloseContainerCommand)command).getProto();
     final ContainerController controller = ozoneContainer.getController();
     final long containerId = closeCommand.getContainerID();
+    if (LOG.isDebugEnabled()) {
```
Comment (bharatviswa504) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/CloseContainerCommandHandler.java:80]: I don't think we need isDebugEnabled

**Review by bharatviswa504 — CHANGES_REQUESTED** [review]
LGTM. 
Few minor comments

**Comment by bharatviswa504** [comment]
@kerneltime Test failure looks related, i think we need to change to markContainerForClose.

**Review by bharatviswa504 — APPROVED** [review]
+1 LGTM

**Comment by kerneltime** [comment]
@bharatviswa504 please take a look now. The integrations tests are stable now including in my sandbox.

**Comment by bharatviswa504** [comment]
Thank You @kerneltime for the contribution.

**Pull request merged**